hudi-cli/src/main/java/org/apache/hudi/cli/commands/HoodieLogFileCommand.java

```diff
@@ -215,9 +215,11 @@ public String showLogFileRecords(
           if (n instanceof HoodieAvroDataBlock) {
             HoodieAvroDataBlock blk = (HoodieAvroDataBlock) n;
             List<IndexedRecord> records = blk.getRecords();
-            allRecords.addAll(records);
-            if (allRecords.size() >= limit) {
-              break;
+            for (IndexedRecord record : records) {
+              if (allRecords.size() >= limit) {
```

Comment:
nitpick: reverse the if condition and avoid 'break'

Reply:
Reasonable, 👍